pkg/common/proto/agent.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+service AgentService{
+    rpc PerformFantasyActions(FantasyLandscape) returns (FantasySelections) {}
+}
+
+message FantasyLandscape{
+    int32 matchNumber = 1;
+    LeagueSettings settings = 2;
+    FantasyTeam botTeam = 3;
+    repeated Player players = 4;
+    /* TODO historical / statistical data */
+    /* TODO competition lineups */
+}
+
+message FantasySelections{
+    repeated PlayerSlot slots = 1;
+}
+
+message FantasyTeam{
+    string id = 1;
+    string name = 2;
+    string owner = 3;
+}
+
+message LeagueSettings{
+    int32 numTeams = 1;
+    repeated PlayerSlot slotsPerTeam = 2;
+    /* scoring, budgets, etc. */
+}
+
+message PlayerSlot{
+    string name = 1;
+    bool canBeEmpty = 2;
+    repeated string allowedPlayerPositions = 4;
+    optional string assignedPlayerId = 3;
```

Comment:
This is slightly out of my wheelhouse but I *think* that optional fields are fairly unnecessary in proto3.

I think it just slightly changes the serialization behavior: https://protobuf.dev/programming-guides/proto3/#field-labels

I.e. the optional field behaves identically from a code standpoint. The go codegen at least does not make a distinction.
 https://protobuf.dev/programming-guides/proto3/#field-labels

Reply:
it actually seems like the community is pretty torn on this one.. I'm not a proto expert, but it does feel like it makes it cleaner to specify when we expect it to be optional vs. not (at least from a business logic standpoint).

Also looks like it may be used to generate methods to check existence? https://stackoverflow.com/questions/42622015/how-to-define-an-optional-field-in-protobuf-3